tests/core/mountpoint/test_base.py

```diff
@@ -579,22 +580,26 @@ async def test_cancel_mount_workspace(base_mountpoint, alice_user_fs, event_bus,
     """
     wid = await alice_user_fs.workspace_create("w")
 
-    # The timeout for `_stop_fuse_thread` is 1 second (3 seconds for macOS),
-    # so let's use a slightly lower timeout to make sure a potential failure
-    # doesn't go undetected.
-    timeout = 3.0 if sys.platform == "darwin" else 1.0
-    with trio.fail_after(timeout * 0.9):
+    # Reuse the same mountpoint manager for all the mountings to
+    # make sure state is not polutated by previous mount attempts
+    async with mountpoint_manager_factory(
+        alice_user_fs, event_bus, base_mountpoint
+    ) as mountpoint_manager:
+
+        for timeout in count(0, 0.002):
```

Comment:
Using `itertools.count` plus a break once the workspace has been mounted is a very neat idea :+1: